### ISSUE TITLE
How to publish battery stats to mqtt msg?

### ISSUE THREAD

**Issue by kevinshane** [RiRomain/Happy_Bubble_ESP32_Node#2]
Hi Sir, first of all very awesome jobs, thanks for sharing the codes
I have a nut mini tracker which has a battery left status, how to add this into mqtt msg?
```
---------------- Found new device  ------------------Advertised Device: Name: nut, Address: cf:51:a4:d8:a7:59, manufacturer data: 59000018, serviceUUID: 00001803-0000-1000-8000-00805f9b34fb 
Address          :cf:51:a4:d8:a7:59
RSSI             : -74
Device name      : nut
ManufacturerData : Y
ServiceUUID      : 00001803-0000-1000-8000-00805f9b34fb
```
Also, could u please add the possibility to define a known mac device so that
1. When it detects and match the "pre-defined" mac in code, it connects to wifi and publish a msg say hey I can saw this device, here is the RSSI, battery stats, serviceUUID etc....
2. When not seeing a "pre-defined" mac address, it stay calm and do nothing except keep scaning to find any device

Also another big problem is the code keep running around 5 or 10 minutes and suddently can't connect to the mqtt server, and the esp32 keeps trying to connect but never success, what cause this problem? 
```
Attempting MQTT connection...failed, rc=-2 try again in 5 seconds
```
After I re-power the chip, it can't find the device anymore, not sure what happen
```
New loop
[0;31mE (164336) BT: btc_search_callback  BLE observe complete. Num Resp 0
[0m

Total devices found: 0
Scan done!
New loop
```
Could u please help me? Thanks!

**Comment by RiRomain** [comment]
> After I re-power the chip, it can't find the device anymore, not sure what happen

Yes, that's a known problem, we need to use the 2 cpu to handle the bluetooth detection using another cpu.
I'm still working on a new version, but don't really have too much time at hand for now...
